app/scripts/metamask-controller.js

```diff
@@ -1317,6 +1317,13 @@ export default class MetamaskController extends EventEmitter {
         accounts = await keyringController.getAccounts();
       }
 
+      // This must be set as soon as possible to communicate to the
+      // keyring's iframe and have the setting initialized properly
+      // Optimistically called to not block Metamask login due to
+      // Ledger Keyring GitHub downtime
+      const transportPreference = this.preferencesController.getLedgerTransportPreference();
+      this.setLedgerTransportPreference(transportPreference);
```

Comment:
Could this ever be `undefined`?  Should we set a default in `setLedgerTransportPreference`?

Reply:
there is a default set in the preferences controller, so I don't think this could be undefined